App.tsx

```diff
@@ -1,22 +1,83 @@
-import { StatusBar } from "expo-status-bar";
-import React from "react";
-import { StyleSheet, Text, View } from "react-native";
-import CameraView from "./components/CameraView";
-
-export default function App(): JSX.Element {
-	return (
-		<View style={styles.container}>
-			<CameraView />
-			<StatusBar style="auto" />
-		</View>
-	);
-}
+import React, { useEffect, useState } from "react";
+import { NavigationContainer } from "@react-navigation/native";
+import { createStackNavigator } from "@react-navigation/stack";
+import CameraScreen from "./screens/CameraScreen";
+import TranscriptionScreen from "./screens/TranscriptionScreen";
+import * as cocoSsd from "@tensorflow-models/coco-ssd";
+import * as tf from "@tensorflow/tfjs";
+import "@tensorflow/tfjs-react-native";
+import { Text, StyleSheet, View } from "react-native";
+
+export type RootStackProps = {
+	Camera: {
+		filePath: string | undefined;
+		objectModel: cocoSsd.ObjectDetection;
+	};
+	Transcription: {
+		latestImagePath: string;
+		objectModel: cocoSsd.ObjectDetection;
+	};
+};
+
+const RootStack = createStackNavigator<RootStackProps>();
+
+const App = (): JSX.Element => {
+	const [objDetectModel, setObjDetectModel] = useState<cocoSsd.ObjectDetection>();
+	const [canUseCamera, setCanUseCamera] = useState<boolean>(false);
+	const [status, setStatus] = useState<string>("Loading tensorflow...");
+
+	useEffect(() => {
+		const load = async () => {
+			await tf.ready();
+
+			setStatus("Loading object detection model...(app will be locked until loaded)");
+
+			if (!objDetectModel) {
+				const objectDetection = await cocoSsd.load();
+				setObjDetectModel(objectDetection);
+				setCanUseCamera(true);
+			}
+		};
+
+		load().catch(err => {
+			throw err;
+		});
+	}, []);
+
+	if (!canUseCamera) {
+		return (
+			<View style={styles.container}>
+				<Text>{status}</Text>
+			</View>
+		);
+	} else {
+		return (
+			<NavigationContainer>
+				<RootStack.Navigator initialRouteName="Camera">
+					<RootStack.Group screenOptions={{ headerShown: false }}>
+						<RootStack.Screen
+							name="Camera"
+							component={CameraScreen}
+							initialParams={{ objectModel: objDetectModel }}
```

Comment:
Why does the camera screen nee the obj detection model if it's not doing any processing?